Add a listAliases muxrpc endpoint

boreq opened the pull request:
I would like to get some feedback on adding a function similar to this one. This would make it possible to retrieve a list of all aliases for a particular user. This list could be used by users to preview their aliases without using the web interface.

boreq commented:
I was considering making this a source request so that a lot of results could be returned efficiently. I don't know if this would make sense.

staltz commented:
Hey @boreq, this PR seems fine, but it would need tests. I haven't worked directly on muxrpc APIs in go-ssb, but the code looks fine in my eyes. I'll still check if others Gophers have time to review this: @decentral1se @cblgh .

boreq commented:
> Hey @boreq, this PR seems fine, but it would need tests.

Sure, I just wanted to get feedback if this would make any sense, the code was broken actually. I just added tests and fixed the code.

decentral1se commented on muxrpc/handlers/alias/handler.go:
```diff
@@ -147,3 +147,47 @@ func (h Handler) Revoke(ctx context.Context, req *muxrpc.Request) (interface{},
+		if errors.As(err, &takenErr) {
+			return nil, takenErr
+		}
+		return nil, fmt.Errorf("listAlias: could not register alias: %w", err)
```
We're not trying to register, we're trying to list?

boreq replied:
Right, this error handling in general was just bollocks, I removed it. Sorry, I whipped this out pretty quickly. There are some mistakes like this with error strings in other handlers as well which I will fix in a separate PR.

boreq replied:
6b932f2

decentral1se replied:
Nice one, thanks!

decentral1se reviewed (commented):
LGTM! Nice one :rainbow:

staltz commented:
Thanks for the PR and thanks for the review!
